### PR TITLE
Initial v1.0.0

### PR DESCRIPTION
**Pull request by Crypto2099**
Initial version of the pure-PHP Bech32 Encoder/Decoder for Cardano PHP. This version is capable of encoding/decoding Cardano Addresses. Adding other encoding/decoding will be added as separate issues and feature developments.

### CODE REVIEW

File: src/Bech32.php [hunk 1]
```diff
@@ -0,0 +1,588 @@
...
+        19, -1, 1, 0, 3, 16, 11, 28, 12, 14, 6, 4, 2, -1, -1, -1, -1, -1,
+    ];
+
+    /**
+     * @param string $hrp
+     * @param array  $combinedDataChars
+     *
+     * @return string
+     */
+    public static function encode(string $hrp, array $combinedDataChars): string
```
Comment (latheesan-k) [inline src/Bech32.php:44]: Minor: The `encode` method does not handle empty data arrays explicitly. While the current tests do not cover this, it should be ensured that encoding with an empty data array results in a valid Bech32 string with only the HRP and separator.

File: src/Bech32.php [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+     * Validates a Bech32 string and returns [$hrp, $dataChars] if
+     * the conversion was successful. An exception is thrown on invalid
+     * data.
+     *
+     * @param string $sBech - the Bech32 encoded string
+     *
+     * @return array - returns [$hrp, $dataChars]
+     * @throws Exception
+     */
+    public static function decode(string $sBech): array
```
Comment (latheesan-k) [inline src/Bech32.php:67]: Minor: The `decode` does not validate that the HRP conforms to Bech32 character constraints (a-z, 0-9). This could lead to issues if an HRP with invalid characters is used.

Potential validation could be done using the following regex pattern `/^[a-z0-9]+$/`

File: tests/Bech32Test.php [hunk 3]
```diff
@@ -0,0 +1,297 @@
+<?php
+
+namespace CardanoPhp\Bech32;
+
+use Exception;
+use PHPUnit\Framework\Attributes\CoversClass;
+use PHPUnit\Framework\TestCase;
+
+#[CoversClass(Bech32::class)]
+class Bech32Test extends TestCase
```
Comment (latheesan-k) [inline tests/Bech32Test.php:10]: Minor: 

* Add Tests for Empty Data Arrays:
  * Ensure that encoding with empty data arrays produces the expected Bech32 string format.

* Enhance HRP Validation:
  * Implement and test checks to ensure that the HRP contains only valid Bech32 characters.

**Review by latheesan-k — APPROVED** [review]
LGTM, some minor recommendations

**Review by latheesan-k — APPROVED** [review]

File: src/Bech32.php (hunk 1 above)
Comment (Crypto2099) [reply src/Bech32.php:44]: This has been added as a new issue and marked as a "Good First Issue" for a contributor

File: src/Bech32.php (hunk 2 above)
Comment (Crypto2099) [reply src/Bech32.php:67]: This has been created as a new issue and marked as a "Good First Issue" for a contributor

File: tests/Bech32Test.php (hunk 3 above)
Comment (Crypto2099) [reply tests/Bech32Test.php:10]: This has been created as a new issue and marked as a "Good First Issue" for a contributor
